corehq/apps/app_manager/static/app_manager/js/graph-config.js

```diff
@@ -383,12 +384,15 @@ var GraphViewModel = function(moduleOptions){
      * of the view model.
      */
     self.getSeriesConstructor = function(){
-        if (self.selectedGraphType() == "xy" || self.selectedGraphType() == "time"){
-            return XYGraphSeries;
-        } else if (self.selectedGraphType() == "bubble"){
+        if (!_.contains(self.availableGraphTypes(), self.selectedGraphType())){
+            throw "Invalid selectedGraphType";
+        }
+        if (self.selectedGraphType() === "bubble"){
```

Comment:
Could, though I'm not certain it would be an improvement. Any more thoughts on how that would be cleaner / more concise / etc.?